app/services/browse_everything_session/provider_session.rb

```diff
@@ -0,0 +1,41 @@
+# Object for handling session cookies containing cached values
+class BrowseEverythingSession
+  class ProviderSession < Base
+    class_attribute :sessions
+    self.sessions = {}
+
+    def self.for(session:, name:)
+      return sessions[name] if sessions[name]
+      sessions[name] = ProviderSession.new(session: session, name: name)
+    end
+
+    def initialize(session:, name:)
+      @name = name
+      super(session: session)
+    end
+
+    def token=(value)
+      @session["#{@name}_token"] = value
+    end
+
+    def token
+      @session["#{@name}_token"]
+    end
+
+    def code=(value)
```

Comment:
+1  to code= as a method name.

Reply:
@scherztc this is a Ruby convention for accessor methods.  Normally `attr_writer` or `attr_accessor` or suffice if simple mutation is necessary, but here I was trying to be explicit (and avoid metaprogramming) for the purposes of clarity.